http/http.go

```diff
@@ -35,18 +39,58 @@ func isPrivateIP(ip net.IP) bool {
 	return false
 }
 
-func isLocalAddress(addr string) bool {
-	ip := net.ParseIP(addr)
-	return isPrivateIP(ip)
+type noLocalTransport struct {
+	inner  http.RoundTripper
+	errlog logrus.FieldLogger
 }
 
-// SafeDialContext exchanges a DialContext for a SafeDialContext that will never dial a reserved IP range
-func SafeDialContext(dialContext func(ctx context.Context, network, addr string) (net.Conn, error)) func(ctx context.Context, network, addr string) (net.Conn, error) {
-	return func(ctx context.Context, network, addr string) (net.Conn, error) {
-		if isLocalAddress(addr) {
-			return nil, errors.New("Connection to local network address denied")
-		}
+func (no noLocalTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	ctx, cancel := context.WithCancel(req.Context())
+
+	ctx = httptrace.WithClientTrace(ctx, &httptrace.ClientTrace{
+		ConnectStart: func(network, addr string) {
+			fmt.Printf("Checking network %v\n", addr)
+			host, _, err := net.SplitHostPort(addr)
+			if err != nil {
+				cancel()
+				fmt.Printf("Canceleing dur to error in addr parsing %v", err)
+				return
+			}
+			ip := net.ParseIP(host)
+			if ip == nil {
+				cancel()
+				fmt.Printf("Canceleing dur to error in ip parsing %v", host)
+				return
+			}
+
+			if isPrivateIP(ip) {
+				cancel()
+				fmt.Println("Canceleing dur to private ip range")
```

Comment:
would be great to log these and previous errors on the `errlog` field of the noLocalTransport